src/collapser.js

```diff
@@ -0,0 +1,54 @@
+/**
+ *  Copyright (c) 2016, The Regents of the University of California,
+ *  through Lawrence Berkeley National Laboratory (subject to receipt
+ *  of any required approvals from the U.S. Dept. of Energy).
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree.
+ */
+
+import Processor from "./processor";
+import { isPipeline } from "./pipeline";
+
+/**
+ * A processor which takes a fieldSpec and returns a new event
+ * with a new column that is a collapsed result of the selected
+ * columns. To collapse the columns it uses the supplied reducer
+ * function. Optionally the new column can completely replace
+ * the existing columns in the event.
+ */
+export default class Collapser extends Processor {
+
+    constructor(arg1, options, observer) {
+        super(arg1, options, observer);
+
+        if (arg1 instanceof Collapser) {
+            const other = arg1;
+            this._fieldSpec = other._fieldSpec;
+            this._name = other._name;
+            this._reducer = other._reducer;
+            this._append = other._append;
+        } else if (isPipeline(arg1)) {
+            this._fieldSpec = options.fieldSpec;
+            this._name = options.name;
+            this._reducer = options.reducer;
+            this._append = options.append;
+        } else {
+            throw new Error("Unknown arg to filter constructor", arg1);
```

Comment:
Would it be helpful to make this "Unknown arg to Collapser filter constructor" to give a hint as to where in the pipeline this is blowing up?